src/deploy.js

```diff
@@ -1,51 +1,109 @@
- */
+const validateBuild = (build) => {
+  if (fs.existsSync(build)) {
+    console.log(`info: validated build directory ${build}`);
+    return true;
+  }
+  console.log(`error: build directory '${build}' does not exist`);
+  return false;
+};
 
-const deploy = async (opts) => {
-  // const { path } = opts;
+const isDefined = (variable) => {
+  return typeof variable !== 'undefined';
+};
 
-  const usageMessage = console.log(
-    'usage: $0 [-e <path/to/file>] [-v <version string>] [-b <path/to/build>]',
-  );
+const deploy = async (opts) => {
+  const { tag, env, build } = opts;
 
-  console.log(`Exectued with path: ${opts.path}`);
-  console.log(usageMessage);
+  // Validate command options
+  if (!validateTag(tag) || !validateEnv(env) || !validateBuild(build)) {
+    console.error('Abort...');
+    return false;
+  }
 
   // fetch environment variables
-  dotenv.config({ path: path.resolve(process.cwd(), '.env.dev') });
-  /* eslint-disable no-unused-vars */
+  // dotenv.config({ path: path.resolve(process.cwd(), '.env.dev') });
+  dotenv.config({ path: env });
   const {
     REACT_APP_GRAASP_DEVELOPER_ID,
     REACT_APP_GRAASP_APP_ID,
-    REACT_APP_GRAASP_DOMAIN,
     REACT_APP_HOST,
     REACT_APP_VERSION,
-    REACT_APP_BASE,
-    NODE_ENV,
     BUCKET,
-    AWS_DEFAULT_REGION,
     AWS_ACCESS_KEY_ID,
     AWS_SECRET_ACCESS_KEY,
     DISTRIBUTION,
   } = process.env;
-  /* eslint-enable no-unused-vars */
 
-  AWS.config.getCredentials(function (err) {
-    if (err) console.error(err.stack);
-    // credentials not loaded
-    else {
-      console.log('Access key:', AWS.config.credentials.accessKeyId);
+  // ensure the correct app variables are defined
+  if (
+    !isDefined(REACT_APP_HOST) ||
+    !isDefined(REACT_APP_GRAASP_DEVELOPER_ID) ||
+    !isDefined(REACT_APP_GRAASP_APP_ID)
+  ) {
+    console.error(
+      'error: environment variables REACT_APP_GRAASP_APP_ID, REACT_APP_GRAASP_DEVELOPER_ID and/or REACT_APP_HOST are not defined',
+    );
+    console.error(
+      'error: you can specify them through a .env file in the app root folder',
+    );
+    console.error('error: or through another file specified with the -e flag');
+    return false;
+  }
+
+  // ensure the correct aws credentials are defined
+  if (
+    !isDefined(BUCKET) ||
+    !isDefined(AWS_ACCESS_KEY_ID) ||
+    !isDefined(AWS_SECRET_ACCESS_KEY)
+  ) {
+    console.error(
+      'error: environment variables BUCKET, AWS_ACCESS_KEY_ID and/or AWS_SECRET_ACCESS_KEY are not defined',
+    );
+    console.error(
+      'error: make sure you setup your credentials file correctly using the scripts/setup.sh script',
+    );
+    console.error(
+      'error: and contact your favourite Graasp engineer if you keep running into trouble',
+    );
+  }
+
+  console.log(
+    `info: publishing app ${REACT_APP_GRAASP_APP_ID} version ${REACT_APP_VERSION}`,
+  );
+
+  // configure the deployment
+  AWS.config.getCredentials((err) => {
```

Comment:
I think you need to `promisify` and `await` here, no?

Reply:
@juancarlosfarah Absolutely correct. But currently not enough time to include it in this PR. Consequently, the commands risk not to be executed in the correct chronological order.